docs/integrations/azure/org.md

```diff
@@ -258,21 +258,96 @@ The `myUserTransformer`, `myGroupTransformer`, `myOrganizationTransformer`, and
+
 ```ts title="packages/backend/src/transformers.ts"
 import * as MicrosoftGraph from '@microsoft/microsoft-graph-types';
 import {
   defaultGroupTransformer,
   defaultUserTransformer,
   defaultOrganizationTransformer,
+  microsoftGraphOrgEntityProviderTransformExtensionPoint,
   MicrosoftGraphProviderConfig,
 } from '@backstage/plugin-catalog-backend-module-msgraph';
 import { GroupEntity, UserEntity } from '@backstage/catalog-model';
+import { createBackendModule } from '@backstage/backend-plugin-api';
+
+// The Group transformer transforms Groups that are ingested from MS Graph
+export async function myGroupTransformer(
+  group: MicrosoftGraph.Group,
+  groupPhoto?: string,
+): Promise<GroupEntity | undefined> {
+  const backstageGroup = await defaultGroupTransformer(group, groupPhoto);
+  return backstageGroup;
+}
+
+// The User transformer transforms Users that are ingested from MS Graph
+export async function myUserTransformer(
+  graphUser: MicrosoftGraph.User,
+  userPhoto?: string,
+): Promise<UserEntity | undefined> {
+  const backstageUser = await defaultUserTransformer(graphUser, userPhoto);
+  return backstageUser;
+}
+
+// The Organization transformer transforms the root MS Graph Organization into a Group
+export async function myOrganizationTransformer(
+  graphOrganization: MicrosoftGraph.Organization,
+): Promise<GroupEntity | undefined> {
+  const backstageOrg = await defaultOrganizationTransformer(graphOrganization);
+  return backstageOrg;
+}
+
+// The Provider Config transformer enables modification of the plugin config
+export async function myProviderConfigTransformer(
+  provider: MicrosoftGraphProviderConfig,
+): Promise<MicrosoftGraphProviderConfig> {
+  return provider;
+}
+
+// Wrapping these functions in a Module allows us to inject them into the Catalog plugin easily
+export const myMsgraphTransformersModule = createBackendModule({
+  pluginId: 'catalog',
+  moduleId: 'msgraph-org',
+  register(reg) {
+    reg.registerInit({
+      deps: {
+        microsoftGraphTransformers:
+          microsoftGraphOrgEntityProviderTransformExtensionPoint,
+      },
+      async init({ microsoftGraphTransformers }) {
+        // Set the transformers to our custom functions
+        microsoftGraphTransformers.setUserTransformer(myUserTransformer);
+        microsoftGraphTransformers.setGroupTransformer(myGroupTransformer);
+        microsoftGraphTransformers.setOrganizationTransformer(
+          myOrganizationTransformer,
+        );
+        microsoftGraphTransformers.setProviderConfigTransformer(
+          myProviderConfigTransformer,
+        );
+      },
+    });
+  },
+});
+
+// Export a default to make importing into the backend simpler
+export default myMsgraphTransformersModule;
+```
 
-// This group transformer completely replaces the built in logic with custom logic.
+Now lets customize each of the providers to suit our needs.
```

Comment:
For the User example I get the why you would do what's suggested pretty easily, but for the others - Group, Organization, and Config - it's less clear to me. Could we maybe include a bit about why you might want to do what's suggested in each of the examples? That could help Adopters get a better feel for which of these might help them with their Use Case.

Reply:
I've added some reasoning to the other providers, but I'm unsure about the ConfigTransformer. The reasoning seems like it should just be added to the config filter itself, but I couldn't think of a 'proper' use for the ConfigTransformer without getting too technical.